File: solutions/ruby/triangle/2/triangle.rb
class Triangle
  def initialize(sides)
    @sides = sides
  end

  def equilateral?
    @sides.uniq.length == 1 unless @sides[0].zero?
  end

  def isosceles?
    @sides.uniq.length <= 2 unless @sides.count(@sides.min) == 2
  end

  def scalene?
    @sides.uniq.length == 3 unless inequality?
  end

  private
  def inequality?
    a, b, c = @sides
    a + b < c or b + c < a or a + c < b
  end
end

File: solutions/python/triangle/2/triangle.py
def equilateral(sides: list[int]) -> bool:
    return len(set(sides)) == 1 and sides[0] != 0


def isosceles(sides: list[int]) -> bool:
    return sides.count(max(sides)) >= 2


def scalene(sides: list[int]) -> bool:
    return len(set(sides)) == 3 and max(sides) <= sum(set(sides) - set([max(sides)]))
